Detectors/TRD/base/include/TRDBase/TrackletTransformer.h

```diff
@@ -0,0 +1,51 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef O2_TRD_TRACKLETTRANSFORMER_H
+#define O2_TRD_TRACKLETTRANSFORMER_H
+
+#include "TRDBase/Geometry.h"
+
+namespace o2
+{
+namespace trd
+{
+
+class TrackletTransformer
+{
+ public:
+  TrackletTransformer();
+  ~TrackletTransformer() = default;
+
+  void loadPadPlane(int hcid);
+
+  float calculateX();
+
+  float calculateY(int hcid, int column, int position);
+
+  float calculateZ(int padrow);
+
+  float calculateDy(int slope);
+
+  float calibrateX(double x, double t0Correction);
+
+  float calibrateDy(double rawDy, double oldLorentzAngle, double lorentzAngle, double driftVRatio);
+
+  std::array<float, 3> transformL2T(int hcid, std::array<double, 3> spacePoint);
+
```

Comment:
As mentioned below, I think it would be elegant to have a public interface that takes a Tracklet64, knows everything about calibration and geometry (including alignment) and then spits out a Calibrated Tracklet.

Reply:
Okay, so I understand that there should be a function here that can transform and calibrate tracklets totally independently of the workflow.